src/Compilers/CSharp/Portable/Binder/Semantics/Conversions/ConversionsBase.cs

```diff
@@ -1636,14 +1636,23 @@ internal static CollectionExpressionTypeKind GetCollectionExpressionTypeKind(CSh
                     return CollectionExpressionTypeKind.Array;
                 }
             }
-            else if (isSpanType(compilation, destination, WellKnownType.System_Span_T, out elementType))
+            else if (isSpanOrListType(compilation, destination, WellKnownType.System_Collections_Immutable_ImmutableArray_T, out elementType)
```

Comment:
>isSpanOrListType [](http://example.com/codeflow?start=21&length=16)

nit: should we find a (yet) more generic name for this local function `isSpanOrListType`? Maybe `isGenericTypeOfT`?

Reply:
I think that would be fine, but I'd like to minimize churn where possible in the immediate term.